Move psychologist immunity from mind to skillchip

coiax opened the pull request:
:cl: coiax
tweak: Psychologist immunity to supermatter hallucinations, and the
ability to make it "calmer", is now provided by a skillchip, rather than
innate.
/:cl:

Skillchips are neat, and this is the sort of "job-specific skill" that
can remain job specific, but still gainable in the shift through
in-universe means.

The logic behind the skillchip is that the supermatter is psychically
sensitive, hence the hallucinations and headaches from looking at it,
and projecting the "correct mental aura" will help it calm down. But
that is lore following mechanics.

- Refactors skillchips to be able to automatically apply more than one
  trait.

coiax committed:
Move psychologist immunity from mind to skillchip

:cl: coiax
tweak: Psychologist immunity to supermatter hallucinations, and the
ability to make it "calmer", is now provided by a skillchip, rather than
innate.
/:cl:

Skillchips are neat, and this is the sort of "job-specific skill" that
can remain job specific, but still gainable in the shift through
in-universe means.

The logic behind the skillchip is that the supermatter is psychically
sensitive, hence the hallucinations and headaches from looking at it,
and projecting the "correct mental aura" will help it calm down. But
that is lore following mechanics.

interestingusernam3 commented on code/modules/library/skill_learning/skillchip.dm:
```diff
@@ -7,8 +7,8 @@
-	/// Trait automatically granted by this chip, optional
-	var/auto_trait
+	/// Traits automatically granted by this chip, optional
+	var/list/auto_traits
```
Replacement of auto_trait with auto_traits should probably be mentioned in the PR description.

Rohesie commented on code/modules/library/skill_learning/skillchip.dm:
```diff
@@ -142,8 +142,9 @@
 
-	if(auto_trait)
-		ADD_TRAIT(user, auto_trait, SKILLCHIP_TRAIT)
+	if(auto_traits)
+		for(var/trait in auto_traits)
+			ADD_TRAIT(user, trait, SKILLCHIP_TRAIT)
```
```suggestion
	for(var/trait in auto_traits)
		ADD_TRAIT(user, trait, SKILLCHIP_TRAIT)
```
No need to check if the list exists.

Rohesie commented on code/modules/library/skill_learning/skillchip.dm:
```diff
@@ -176,8 +177,9 @@
 
-	if(auto_trait)
-		REMOVE_TRAIT(user, auto_trait, SKILLCHIP_TRAIT)
+	if(auto_traits)
+		for(var/trait in auto_traits)
+			REMOVE_TRAIT(user, trait, SKILLCHIP_TRAIT)
```
```suggestion
	for(var/trait in auto_traits)
		REMOVE_TRAIT(user, trait, SKILLCHIP_TRAIT)
```

Rohesie commented on code/modules/library/skill_learning/skillchip.dm:
```diff
@@ -7,8 +7,8 @@
 
-	/// Trait automatically granted by this chip, optional
-	var/auto_trait
+	/// Traits automatically granted by this chip, optional
```
```suggestion
	/// Traits automatically granted by this chip, optional. Lazy list.
```

coiax committed:
Merge branch 'master' into hypergiga-skillchip

coiax committed:
Improve documentation and lazy list handling

coiax committed:
Modify merged skillchip with new list format

The "nuclear authentication disk" identification skillchip has been
added since this PR was originally opened, need to update it, since
skillchips can now give a list of traits.
